src/txn/txn_recover.c

```diff
@@ -452,6 +452,67 @@ __recovery_set_oldest_timestamp(WT_RECOVERY *r)
     return (0);
 }
 
+/*
+ * __recovery_set_checkpoint_snapshot --
+ *     Set the checkpoint snapshot details as retrieved from the metadata file.
+ */
+static int
+__recovery_set_checkpoint_snapshot(WT_RECOVERY *r)
+{
+    WT_CONFIG_ITEM cval;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    char *sys_config;
+    const char *sep;
+
+    sys_config = NULL;
+
+    session = r->session;
+    conn = S2C(session);
+
+    /*
+     * Read the system checkpoint information from the metadata file and save the snapshot related
+     * details of the last checkpoint for later query. This gets saved in the connection.
+     */
+
+    /* Search the metadata for the system information. */
+    WT_ERR_NOTFOUND_OK(
+      __wt_metadata_search(session, WT_SYSTEM_CKPT_SNAPSHOT_URI, &sys_config), false);
+    if (sys_config != NULL) {
+        WT_CLEAR(cval);
+        WT_ERR_NOTFOUND_OK(
+          __wt_config_getones(session, sys_config, WT_SYSTEM_CKPT_SNAPSHOT_MIN, &cval), false);
```

Comment:
You are searching for each entry, and then setting values. Did you think about instead doing a traversal through the config and setting based on the ones you find?

This code isn't particularly performance sensitive, but the latter approach feels more efficient to me.

I also wonder whether we need to require any of the fields being present? i.e: I presume it'd be bad if the snapshot had specific transaction IDs in it, if there was no max.
I also think it'd be nice to confirm that the `snapshot_count` corresponds to the number of entries we retrieved from the list.

Reply:
Added validation check to return an error if any of the fields not present after finding `system:checkpoint_snapshot` in metadata file and to confirm the number of entries retrieved matches with `snapshot_count `.